MSAL/src/MSALErrorConverter.m

```diff
@@ -0,0 +1,80 @@
+//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+#import "MSALErrorConverter.h"
+#import "MSALError_Internal.h"
+
+static NSDictionary* s_errorDomainMapping;
+static NSDictionary* s_errorCodeMapping;
+
+@implementation MSALErrorConverter
+
++ (void)initialize
+{
+    s_errorDomainMapping = @{
+                             MSIDErrorDomain : MSALErrorDomain
+                             };
+    
+    s_errorCodeMapping = @{
+                           //sample format is like @"MSIDErrorDomain|-10000":@"MSALErrorDomain:-20000"
```

Comment:
I'd prefer using the same delimiter, e.g. | in both key and value

Reply:
Another question is why are both domain and code necessary in the value part? I agree that it's necessary for the key, but for domain mapping there's already s_errorDomainMapping. Another alternative is to remove s_errorDomainMapping and get domain from this mapping.

Reply:
Agree.

Regarding the alternative, current design can work for the scenario where we only want to map the domain but keep the error code, like NSOSStatusErrorDomain->ADKeychainErrorDomain. For such a scenario, we only need to fill up s_errorDomainMapping but no need to fill up  s_errorCodeMapping, what do you think?

Reply:
I agree, it makes sense to keep s_errorDomainMapping. Just let's remove the domain from the value for s_errorCodeMapping unless there's an actual use for it. Thanks :)